src/main/java/com/paskar/email/application/console/InformationFromConsole.java

```diff
@@ -0,0 +1,117 @@
+package com.paskar.email.application.console;
+
+
+import lombok.Getter;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class InformationFromConsole implements Serializable {
+
+    @Getter
+    private List<Email> listWithAllEmails = new ArrayList<>();
+
+    private static final long serialVersionUID = 1L;
+
+    private transient static final DateTimeFormatter FORMATTER =
+            DateTimeFormatter.ofPattern("MM dd yyyy HH:mm");
+    private transient static final BufferedReader READER = new BufferedReader(new InputStreamReader(System.in));
+
+
+    public static void main(String[] args) throws IOException {
+        InformationFromConsole console = new InformationFromConsole();
+
+        int numberOfLetters = numberOfLettersValidation();
+
+        for (int i = 0; i < numberOfLetters; i++) {
+            console.getListWithAllEmails().add(createNewEmail());
+        }
+
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream("emailList.bin"))) {
```

Comment:
Don't use native java serialization. It will be deprecated and removed in future releases https://cr.openjdk.java.net/~briangoetz/amber/serialization.html. You can use simple text format, like CSV, instead. And you need to extract this code that deals with emails storage into separate component to be able to replace it with db access in future. You can define an interface like:
```
public interface EmailRepository {
   void save(Email email);
   List<Email> findAll();
   List<Email> findEmailsNearDeliveryDate(LocalDateTime currentDate);
}
```
and then implement this interface using serialization(not java native) technique to file. Then simply inject EmailRepository as dependency in this component(and the one with @Scheduled). Spring will automatically pick up implementation and make injection happen